dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -63,13 +71,68 @@ def __add__(self, other: CategoricalColumn) -> CategoricalColumn:
-        merged_profile._categories = utils.add_nested_dictionaries(
-            self._categories, other._categories
-        )
-        BaseColumnProfiler._add_helper(merged_profile, self, other)
-        self._merge_calculations(
-            merged_profile.__calculations, self.__calculations, other.__calculations
-        )
+
+        # If both profiles have not met stop condition
+        if not (self._stop_condition_is_met or other._stop_condition_is_met):
+            merged_profile._categories = utils.add_nested_dictionaries(
+                self._categories, other._categories
+            )
+
+            BaseColumnProfiler._add_helper(merged_profile, self, other)
+            self._merge_calculations(
+                merged_profile.__calculations, self.__calculations, other.__calculations
+            )
+
+            # Transfer stop condition variables of 1st profile object to merged profile
+            # if they are not None else set to 2nd profile
+            if self.max_sample_size_to_check_stop_condition is not None:
+                merged_profile.max_sample_size_to_check_stop_condition = (
+                    self.max_sample_size_to_check_stop_condition
+                )
+            else:
+                merged_profile.max_sample_size_to_check_stop_condition = (
+                    other.max_sample_size_to_check_stop_condition
+                )
+            if self.stop_condition_unique_value_ratio is not None:
+                merged_profile.stop_condition_unique_value_ratio = (
+                    self.stop_condition_unique_value_ratio
+                )
+            else:
+                merged_profile.stop_condition_unique_value_ratio = (
+                    other.stop_condition_unique_value_ratio
+                )
+
+            # 1st and 2nd profile values should be same so no need for conditonal set
+            merged_profile._stop_condition_is_met = self._stop_condition_is_met
+            merged_profile._stopped_at_unique_ratio = self._stopped_at_unique_ratio
+            merged_profile._stopped_at_unique_count = self._stopped_at_unique_count
+
+            # Check merged profile w/ stop condition
+            if merged_profile._check_stop_condition_is_met(
+                merged_profile.sample_size, merged_profile.unique_ratio
+            ):
+                merged_profile._stopped_at_unique_ratio = merged_profile.unique_ratio
+                merged_profile._stopped_at_unique_count = merged_profile.unique_count
+                merged_profile._categories = {}
+                merged_profile._stop_condition_is_met = True
+
+        else:
+            # If either profile has hit stop condition, remove categories dict
+            merged_profile._stopped_at_unique_ratio = max(
+                self.unique_ratio, other.unique_ratio
+            )
+            merged_profile._stopped_at_unique_count = max(
+                self.unique_count, other.unique_count
+            )
```

Comment:
Should it be purely based on max?

Consider:
* `Profile 1` has 10        samples and a unique_ratio of 1.00 
* `Profile 2` has 10000 samples and a unique ratio of 0.50.

Do these values make sense?

Also, do we want to potentially choose different values from each, could that be problematic?

Reply:
The option of not choosing one profiles values vs the other's complicates the sample_size below I believe.

Reply:
We should also have tests which ensure that if self has the values that get assigned vs swapping and ensuring that other's values can be choosen.